java/io/bazel/rules/closure/webfiles/WebfilesServer.java

```diff
@@ -0,0 +1,277 @@
+import com.google.common.net.MediaType;
+import com.google.template.soy.SoyFileSet;
+import com.google.template.soy.data.SoyListData;
+import com.google.template.soy.data.SoyMapData;
+import com.google.template.soy.tofu.SoyTofu;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.BindException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import java.util.zip.GZIPOutputStream;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.mortbay.jetty.Connector;
+import org.mortbay.jetty.Server;
+import org.mortbay.jetty.bio.SocketConnector;
+import org.mortbay.jetty.servlet.Context;
+import org.mortbay.jetty.servlet.ServletHolder;
+
+/** Development web server for a single webfiles() rule. */
+public final class WebfilesServer extends HttpServlet {
+
+  private static final String BLUE = "\u001b[34m";
+  private static final String BOLD = "\u001b[1m";
+  private static final String RESET = "\u001b[0m";
+
+  private static final HostAndPort DEFAULT_BIND = HostAndPort.fromString("[::]:6006");
+  private static final Path RUNFILES_PATH = Paths.get("..");
+  private static final Path RUNFILES_PREFIX = Paths.get("/_/runfiles");
+  private static final MediaType DEFAULT_MIME_TYPE = MediaType.OCTET_STREAM;
+  private static final Splitter TAB_SPLITTER = Splitter.on('\t');
+  private static final Pattern ALLOWS_GZIP =
+      Pattern.compile("(?:^|,|\\s)(?:(?:x-)?gzip|\\*)(?!;q=0)(?:\\s|,|$)");
+  private static final SoyTofu TOFU =
+      SoyFileSet.builder()
+          .add(getResource(ListingSoyInfo.class, ListingSoyInfo.getInstance().getFileName()))
+          .build()
+          .compileToTofu();
+
+  private static final Logger logger = Logger.getLogger(WebfilesServer.class.getName());
+  private static final HashMap<Path, Path> assets = new HashMap<>();
+  private static final List<Path> reverseTopologicalPaths = new ArrayList<>();
+  private static String label = "//";
+
+  public static void main(String[] args) throws Exception {
+    boolean bindWasAskedFor = false;
+    HostAndPort bind = DEFAULT_BIND;
+    Iterator<String> flags = Iterators.forArray(args);
+    while (flags.hasNext()) {
+      String flag = flags.next();
+      switch (flag) {
+        case "--bind":
+          bind = HostAndPort.fromString(flags.next());
+          bindWasAskedFor = true;
+          break;
+        case "--label":
+          label = flags.next();
+          break;
+        case "--manifest":
+          loadManifest(Paths.get(flags.next()));
+          break;
+        default:
+          throw new RuntimeException("Unexpected flag: " + flag);
+      }
+    }
+    Server server;
+    while (true) {
```

Comment:
The while(true) is so it will keep incrementing ports until it gets one that works?
Maybe comment it or factor it out (if feasible)